geopandas/io/tests/test_io.py

```diff
@@ -17,6 +18,14 @@ def setup_method(self):
             self.crs = f.crs
             self.columns = list(f.meta["schema"]["properties"].keys())
 
+    def test_read_sqlite(self):
+        nybb_sqlite = os.path.join(geopandas.datasets._module_path,
+                                   "nybb.sqlite")
```

Comment:
This is obviously not the intent with `datasets.get_path`. I didn't have any particular inspiration for how to modify the `get_path` handling for a dataset of the same type but in a different format.

Reply:
Also perhaps noteworthy: the nybb sqlite database is >1MB, which might be undesirable enough to suggest alternative handling.

Reply:
If we want to do this, we can use a different name like 'nybb.sqlite'. 
But if it is only for testing, I would maybe not fully include it, certainly if it is quite big.

Can we create it during testing? 

Other alternative is to distribute test data in a different way? (putting them outside the package, or in a different repo)

Reply:
I'll look into generating the sqlite database during testing rather than packaging it. It is ~1 MB, which is comparable to the size of what we're already shipping.